nautobot/core/jobs/cleanup.py

```diff
@@ -65,6 +65,19 @@ def recursive_delete_with_cascade(self, queryset, deletion_summary):
                 related_model = related_object.related_model
                 related_field_name = related_object.field.name
                 cascade_queryset = related_model.objects.filter(**{f"{related_field_name}__id__in": queryset})
+                if cascade_queryset.exists():
+                    self.recursive_delete_with_cascade(cascade_queryset, deletion_summary)
```

Comment:
Note that the `.exists()` check is something I added here and below to help address an infinite-recursion failure that the CI reported. I think it's correct though.